### PR TITLE
`bump_version_update_changelog_create_pr_action`: validate new version is not empty

### PR DESCRIPTION
**Pull request by NachoSoto**
I realized the script kept going after I just pressed "enter" with an empty version number.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Not sure if this is the best way, and also maybe I should add a test for this?

File: lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ def self.run(params)
         # Ask for new version number
         new_version_number ||= UI.input("New version number: ")
 
+        UI.user_error!("Invalid version number: #{new_version_number}") unless !new_version_number.strip.isEmpty?
```
Comment (tonidero) [inline lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb:32]: I think something like this? And yeah, maybe writing a test to make sure it fails in this case would be great 😄 
```suggestion
        UI.user_error!("Version number cannot be empty") if new_version_number.blank?
```

Comment (NachoSoto) [reply lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb]: Blank doesn't exist though?

**Comment by NachoSoto** [comment]
@tonidero updated with tests!

File: lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb (hunk 1 above)
Comment (tonidero) [reply lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb]: Hmm weird... https://apidock.com/rails/Object/blank%3F it's there... Did you add the `?`?

**Comment by tonidero** [comment]
Looks like there is a small linter error, other than that `.strip.empty?` should be the same as `.blank?` so it should work as well. Not sure why `blank?` wouldn't work though...

File: spec/actions/bump_version_update_changelog_create_pr_action_spec.rb [hunk 2]
```diff
@@ -13,6 +13,29 @@
...
+      allow(FastlaneCore::UI).to receive(:interactive?).and_return(true)
+      allow(FastlaneCore::UI).to receive(:confirm).with(anything).and_return(true)
+      allow(FastlaneCore::UI).to receive(:input).with('New version number: ').and_return('')
+      
+      expect(FastlaneCore::UI).to receive(:user_error!)
+        .with('Version number cannot be empty')
+        .once
+        .and_throw(:expected_error)
+
+      catch :expected_error do
```
Comment (tonidero) [inline spec/actions/bump_version_update_changelog_create_pr_action_spec.rb:26]: I've usually catched errors like this: https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/blob/806c2787a95522facb437be5cfad803b74f5e5fe/spec/helper/revenuecat_internal_helper_spec.rb#L100-L102 but this way will also catch the exact error message, so it's more complete 👍

**Review by tonidero — APPROVED** [review]

**Comment by NachoSoto** [comment]
I get this:
```
NoMethodError:
       undefined method `blank?' for "1.13.0":String
       Did you mean?  black
                      black!
```

**Comment by NachoSoto** [comment]
Looks like that's part of `active_support`?

File: lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb (hunk 1 above)
Comment (NachoSoto) [reply lib/fastlane/plugin/revenuecat_internal/actions/bump_version_update_changelog_create_pr_action.rb]: That's in `rails`, but we don't use it?

**Comment by NachoSoto** [comment]
That keeps failing to lint but it's passing locally :/ any idea why? I got the failure locally, I removed that trailing whitespace, and now they pass locally, but still fail in CI.

**Pull request merged**